### PR TITLE
[DTensor][XLA] support XLA backend in distirbute_module API

### PR DESCRIPTION
**Pull request by yeounoh**
Addresses #92909  cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang

### CODE REVIEW

File: torch/distributed/_tensor/_xla.py [hunk 1]
```diff
@@ -197,4 +198,75 @@ def xla_distribute_module(
     input_fn: Optional[Callable[..., None]] = None,
     output_fn: Optional[Callable[..., None]] = None,
 ) -> nn.Module:
-    raise NotImplementedError
+    """
+    This function annotates all module parameters for auto-partitioning with
```
Comment (wanchaol) [inline torch/distributed/_tensor/_xla.py:202]: I remembered we were discussing that we should move the `xla_distribute_tensor` and `xla_distribute_module` implementation to the xla repo directly and only call them in the core APIs, why we didn't do this actually?

Comment (wanchaol) [reply torch/distributed/_tensor/_xla.py:202]: you had a PR on `xla_distribute_tensor` about this already? https://github.com/pytorch/pytorch/pull/113214

Comment (yeounoh) [reply torch/distributed/_tensor/_xla.py:202]: Yes, I moved the implementaiton already to xla, but the refactoring PR hasn't landed. Let's fix it, thanks!

**Comment by yeounoh** [comment]
This depends on https://github.com/pytorch/xla/pull/6683, @wanchaol should we skip `distribute_module` test here, and just rely on the downstream test? I remember that you were suggesting it in the refactoring PR. If we skip here, then this doesn't need to be blocked on https://github.com/pytorch/xla/pull/6683.

**Comment by yeounoh** [comment]
> This depends on [pytorch/xla#6683](https://github.com/pytorch/xla/pull/6683), @wanchaol should we skip `distribute_module` test here, and just rely on the downstream test? I remember that you were suggesting it in the refactoring PR. If we skip here, then this doesn't need to be blocked on [pytorch/xla#6683](https://github.com/pytorch/xla/pull/6683).

I am good keeping it as a safeguard here, to ensure that we can call the xla_distribute_module API via distribute_module here.

File: torch/distributed/_tensor/api.py [hunk 2]
```diff
@@ -673,11 +673,40 @@ def distribute_module(
...
+                xla_distribute_module,
+            )
+
+            return xla_distribute_module(
+                module, device_mesh, partition_fn, input_fn, output_fn
+            )  # type:ignore[return-value]
+        except ImportError as e:
+            msg = "To use DTensor API with xla, you must install the torch_xla package!"
+            raise ImportError(msg) from e
+    if device_type == "xla":
```
Comment (wanchaol) [inline torch/distributed/_tensor/api.py:702]: why there're two if branches? forgot to delete one?

Comment (yeounoh) [reply torch/distributed/_tensor/api.py]: Oh no, yes fixed!

**Review by wanchaol — APPROVED** [review]
lgtm, thanks for addressing comments!

**Comment by yeounoh** [comment]
Not updating the xla pin, since the latest one seems to have some pjrt gpu plugin build issue... cc @vanbasten23

**Comment by yeounoh** [comment]
This is also related to https://github.com/pytorch/xla/issues/6322

**Pull request closed without merging**